nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1005,15 +1244,26 @@ def __init__(
 
         self.fp32_residual_connection = fp32_residual_connection  # if true move residual connections to fp32
 
+        # Layernorm on the input data.
+        if normalization == 'layernorm':
+            self.input_layernorm = get_layer_norm(
+                hidden_size, layernorm_epsilon, persist_layer_norm, sequence_parallel
+            )
```

Comment:
if move this out of `if self.layer_type != LayerType.retrieval_decoder_after_self_attn:` context, it will creates problem (gradient not passing to the self.input_layernorm paramemters) for the LayerType.retrieval_decoder_after_self_attn because it is going to skip the first self attention sublayer and no need of `input_layernorm`.

Reply:
I didn't intend to move it, was just tricky to rebase. We just need it under the conditional?

Reply:
It already has it under the condition, just remove these lines.

Reply:
Looks like this just needed to be removed.